src/main/resources/static/assets/js/sdh.js

```diff
@@ -0,0 +1,68 @@
+(function() {
+	var app = new Vue({
+		el: '#contents',
+		data: {
+			fields: [{key:'domain', sortable:true}, {key:'date_last_assessed', sortable:true}, {key:'clinician_priority', sortable:true}, {key:'patient_readiness', sortable:true}],
+			filters: {
+			      domain: '',
+			      date_last_assessed: '',
+			      clinician_priority: '',
+			      patient_readiness: ''
+			    },
+	        items: [
+	            { domain: 'Food Security', date_last_assessed: '01/01/2018', clinician_priority: 'High', patient_readiness: 'Not Ready' },
+	            { domain: 'Transportation', date_last_assessed: '01/01/2019', clinician_priority: 'Low', patient_readiness: 'Motivated'},
+	            { domain: 'Housing Instability', date_last_assessed: '10/11/2018', clinician_priority: 'Medium', patient_readiness: 'Ready' },
+	            { domain: 'Utilities', date_last_assessed: '02/01/2019', clinician_priority: 'High', patient_readiness: 'Ready' },
+	            { domain: 'Interpersonal Violence', date_last_assessed: '12/01/2018', clinician_priority: 'Medium', patient_readiness: 'Motivated' },
+	            { domain: 'Education', date_last_assessed: '04/04/2018', clinician_priority: 'Medium', patient_readiness: 'Not Ready' },
+	            { domain: 'Financial Strain', date_last_assessed: '03/17/2012', clinician_priority: 'Low', patient_readiness: 'Ready' }
+	          ]
+		},
+		methods: {
+			_priorityVariant (priority) {
+				if (priority == 'High') {
+					return 'danger';
+				} else if (priority == 'Medium') {
+					return 'warning';
+				} else if (priority == 'Low') {
+					return 'success';
+				}
+		    },			
+			_readinessVariant (readiness) {
+				if (readiness == 'Not Ready') {
+					return 'danger';
+				} else if (readiness == 'Motivated') {
+					return 'warning';
+				} else if (readiness == 'Ready') {
+					return 'success';
+				}
+		    }			
+		},
+		computed: {
+		    filtered () {
+		        const filtered = this.items.filter(item => {
+		          return Object.keys(this.filters).every(key =>
+		              String(item[key].toLowerCase()).includes(this.filters[key].toLowerCase()));
+		        });
+		        return filtered.length > 0 ? filtered : [{
+		          domain: '',
+			      date_last_assessed: '',
+		          clinician_priority: '',
+		          patient_readiness: ''
+		        }];
+		    },
+		    emphasized () {
+		    	const { filtered, _priorityVariant, _readinessVariant } = this;
```

Comment:
It doesn't matter in this case, but be careful destructuring methods from an object. It unsets their `this` pointer.

Reply:
OK. Is there a better way to do this? I was essentially following the pattern I was seeing in our other Vue projects. e.g., https://github.com/monarch-initiative/phenotypr-body/blob/master/src/components/FeedbackPage.vue

Reply:
There's nothing wrong with it. I just wanted to point out a possible pitfall of destructuring methods, because it's bitten me multiple times in the past. In this particular case nothing bad will happen, because the methods used are pure functions that never refer to `this`.

Reply:
The Phenotypr example doesn't have this issue, because in that case computed properties are being destructured. Destructuring a CP gets the value of evaluating the CP, whereas destructuring a method gets a reference to the function, without its context.

Now that I've written this, I can see how it could be confusing. There's some magic involved.

Reply:
Ah, that makes sense. Thanks for clarifying.